polyply/src/generate_templates.py

```diff
@@ -267,11 +277,10 @@ def extract_block(molecule, resname, defines):
     # label in the molecule and in the block for
     # relabeling the interactions
     mapping = {}
-    for node in nodes:
+    for node in template_graph.nodes:
         attr_dict = molecule.nodes[node]
-        if attr_dict["resid"] == resid:
-            block.add_node(attr_dict["atomname"], **attr_dict)
-            mapping[node] = attr_dict["atomname"]
+        block.add_node(attr_dict["atomname"], **attr_dict)
+        mapping[node] = attr_dict["atomname"]
```

Comment:
Suggestion: build mapping, and do nx.relabel_nodes to get the Block

Reply:
this function is due for a major overhaul, but I'd like to do that in another PR in particular the gen_ff